### DISCUSSION TITLE
Security implications of a single server deployment

### DISCUSSION THREAD

**Discussion by harichards** [gravitational/teleport#18864] [Q&A] [answered]
Hello,

What would be the security implications of moving from an AWS delpoyment of Teleport with separate proxy and auth nodes, where the auth node is behind a NAT gateway (basically [this](https://github.com/gravitational/teleport/tree/master/examples/aws/terraform/ha-autoscale-cluster#terraform-based-provisioning-example-amazon-single-ami) deployment), to a single server deployment, where the proxy and auth are a single node and not behind a NAT?

What if I remove the use of S3 and DynamoDB and have Teleport store everything locally? Are there any concerns there?

Looking to really slim down my deployment of Teleport (open source) without compromising security too much. I only have a few nodes connected to my cluster.

Thanks!

**Comment by webvictim** [comment, marked as answer]
There is by definition a little more of an attack surface if you put everything on one server, but you can mitigate pretty much all of this by using TLS multiplexing and only opening the Teleport proxy's port 443 to the outside world, keeping everything else firewalled off.

The biggest issue you'd be looking at would probably be a performance decrease based on the fact that everything is going to an on-disk SQLite database now rather than a distributed database engine. Session recordings would likely be better protected by S3 than local storage too, but good controls on the server should prevent issues.

I wouldn't recommend a single-node deployment for use in a production setup, but for a handful of users and services it should work fine.

**Reply by harichards** [reply]
Makes sense. Thanks for your reply!
